### ISSUE TITLE
If I use the value "import logs" in a dict, ansible throws "with_items expects a list" error

### ISSUE THREAD

**Issue by jerrac** [ansible/ansible#6111]
##### Issue Type:

Bug Report
##### Ansible Version:

ansible --version
ansible 1.4.5
##### Environment:

I'm running Ubuntu 12.04 on both testhost and my localhost.

I set `hash_behaviour=merge` in my ansible.cfg file.
##### Summary:

If I use the value "import logs" in a dict, ansible throws "with_items expects a list" error
##### Steps To Reproduce:

This is an odd issue. If I run this playbook:

``` yaml

---
# demonstrate odd error
- hosts: testhost
  vars:
    - common_cronjobs:
        import_logs:
          name: "import logs"
          job: 'echo blah > /dev/null'
          day: "*"
          hour: "0"
          minute: "10"
          weekday: "*"
          state: "present"
  tasks:
    - name: debug common_cronjobs
      debug: msg={{ item }}
      with_items: common_cronjobs|dictsort
```

ansible-playbook will throw an error:

```
TASK: [debug common_cronjobs] ************************************************* 
fatal: [testhost] => with_items expects a list

FATAL: all hosts have already failed -- aborting
```

But, if all I do is change `name: "import logs"` to anything else, like `name: "importlogs"` It works perfectly.
##### Expected Results:

```
PLAY [testhost] ***************************************** 

GATHERING FACTS *************************************************************** 
ok: [testhost]

TASK: [debug common_cronjobs] ************************************************* 
ok: [testhost] => (item=('import_logs', {'name': 'import logs', 'hour': '0', 'state': 'absent', 'job': 'echo blah > /dev/null', 'weekday': '*', 'day': '*', 'minute': '10'})) => {
    "item": [
        "import_logs", 
        {
            "day": "*", 
            "hour": "0", 
            "job": "echo blah > /dev/null", 
            "minute": "10", 
            "name": "import logs", 
            "state": "absent", 
            "weekday": "*"
        }
    ], 
    "msg": "[import_logs,"
}

PLAY RECAP ******************************************************************** 
testhost : ok=2    changed=0    unreachable=0    failed=0   
```
##### Actual Results:

```
PLAY [testhost] ***************************************** 

GATHERING FACTS *************************************************************** 
ok: [testhost]

TASK: [debug common_cronjobs] ************************************************* 
fatal: [testhost] => with_items expects a list

FATAL: all hosts have already failed -- aborting

PLAY RECAP ******************************************************************** 
           to retry, use: --limit @/home/reagand/importerror.retry

testhost : ok=1    changed=0    unreachable=1    failed=0  
```

**Comment by mpdehaan** [comment]
This has nothing to do with the "import logs" but rather that with_items does need a list.

Stop by the mailing list and folks can help with questions.

The fact that debug can't print nested data structures with msg is a thing based on it wanting a quoted string and such, do this instead:

debug: var=varName

**Comment by jerrac** [comment]
> This has nothing to do with the "import logs" but rather that with_items does need a list.

That's why I use dictsort to turn it into a list.

Also, if it was just the list thing, then why does it work when I name it anything except "import logs"?

**Comment by mpdehaan** [comment]
Underlying problems with the "with_items" parsing are addressed in 1.5.4 and the recent devel branch changes.
